packages/server/src/attribute/attribute.service.ts

```diff
@@ -0,0 +1,58 @@
+import { Injectable, HttpStatus, HttpException } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository, getRepository } from 'typeorm';
+import { validate } from 'class-validator';
+
+import { Attribute } from '@attribute/attribute.entity';
+import { CreateAttributeDto } from '@attribute/dto/create-attribute.dto';
+
+@Injectable()
+export class AttributeService {
+  constructor(
+    @InjectRepository(Attribute)
+    private readonly attributeRepository: Repository<Attribute>,
+  ) {}
+
+  async create(dto: CreateAttributeDto): Promise<Attribute> {
+    const { value, name, type } = dto;
+    const qb = await getRepository(Attribute)
+      .createQueryBuilder('attribute')
+      .where('attribute.value = :value', { value });
+
+    const attribute = await qb.getOne();
+
+    if (attribute) {
+      const errors = { value: 'This value already been registered' };
+      throw new HttpException(
+        { message: 'Input data validation failed', errors },
+        HttpStatus.BAD_REQUEST,
+      );
+    }
+
+    const newAttribute = new Attribute();
```

Comment:
What do you think about passing this properties, to the constructor of the `Attribute`?